client/src/app/shared/metadata-form/fields/ontology-input/ontology-input.component.ts

```diff
@@ -0,0 +1,122 @@
+import {Component, OnInit} from '@angular/core';
+import {AbstractControl, FormControl} from '@angular/forms';
+import {concatMap, startWith} from 'rxjs/operators';
+import {Observable} from 'rxjs';
+import {Metadata} from '../../metadata';
+import {OntologyService} from '../../../services/ontology.service';
+import {OlsDoc} from '../../../models/ols';
+import {JsonSchema} from '../../json-schema';
+
+const OLS_RELATION = {
+  'rdfs:subClassOf': 'allChildrenOf'
+};
+
+interface Ontology {
+  text: string;
+  ontology_label: string;
+  ontology: string;
+}
+
+
+@Component({
+  selector: 'app-ontology-input',
+  templateUrl: './ontology-input.component.html',
+  styleUrls: ['./ontology-input.component.css']
+})
+export class OntologyInputComponent implements OnInit {
+  metadata: Metadata;
+  control: AbstractControl;
+  id: string;
+
+  label: string;
+  helperText: string;
+  isRequired: boolean;
+  error: string;
+  example: string;
+  disabled: boolean;
+  value: Ontology;
+
+  searchControl: FormControl;
+  filteredOptions: Observable<Ontology[]>;
+
+  searchParams: object;
+
+  constructor(private ols: OntologyService) {
+  }
+
+  ngOnInit() {
+    const userFriendly = this.metadata.schema.user_friendly ? this.metadata.schema.user_friendly : undefined;
+    this.label = userFriendly ? userFriendly : this.metadata.key;
+
+    const guidelines = this.metadata.schema.guidelines;
+    const description = this.metadata.schema.description;
+    this.helperText = guidelines ? guidelines : description;
+
+    this.isRequired = this.metadata.isRequired;
+    this.example = this.metadata.schema.example;
+    this.disabled = this.metadata.isDisabled;
+
+    this.createSearchParams(this.metadata.schema);
```

Comment:
Could change this to `this.searchParams = createSearchParams(this.metadata.schema)`, would be more declarative and clarify what `searchParams` is, or can be.

Reply:
Same with searchControl